src/Symfony/Component/TypeInfo/Tests/TypeResolver/StringTypeResolverTest.php

```diff
@@ -155,6 +155,7 @@ public static function resolveDataProvider(): iterable
 
         // union
         yield [Type::union(Type::int(), Type::string()), 'int|string'];
+        yield [Type::mixed(), 'int|mixed'];
```

Comment:
maybe test both `int|mixed` and `mixed|int`, to avoid issues where the first or last type would not work (see the discussion comment)